front/src/reducers/osrdconf/index.ts

```diff
@@ -419,15 +420,15 @@ export function updateInfraID(infraID: number | undefined): ThunkAction<ActionUp
       type: UPDATE_INFRA_ID,
       infraID,
     });
-    dispatch(updateSwitchTypes({}));
+    dispatch(updateSwitchTypes([]));
 
     if (infraID) {
       try {
         // get switch types  with rtk query
         const { data: newSwitchTypes } = await dispatch(
           osrdEditoastApi.endpoints.getInfraByIdSwitchTypes.initiate({ id: infraID })
         );
-        dispatch(updateSwitchTypes(newSwitchTypes));
+        dispatch(updateSwitchTypes(newSwitchTypes as SwitchType[]));
```

Comment:
Can you check here that newSwitchTypes is not undefined ? We should update the openApi to clarify what the api returns here